fix: allow gosc to set default gateway on more than one network adapter

spacegospod opened the pull request:
### Description

It appears that `customizations_helper.go` intentionally skips the gateway settings for every adapter after the first one.
This leads to a bug where cloning a VM with multiple adapters ends up with potentially incorrect network settings.

My fix is to simply delete the code which explicitly omits the settings for subsequent adapters during GOS customization.

### Acceptance tests

- [X] Have you added an acceptance test for the functionality being added?
- [X] Have you run the acceptance tests on this branch?

I have added a new testcase that can be used to validate this fix.
The test is intentionally marked with `t.Skip()` so that it does get included in the CI runs.
This is so because it requires a Sysprepped Windows VM template.

I tested this fix by running the new test, logging into the VM and confirming that the gateway is applied to both adapters.

### Release Note

`resource/virtual_machine`: Allow `ipv4_gateway` to be set on more than one network adapter when cloning a virtual machine.

### References

Closes #1232

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
+
+  num_cpus = 4
+  memory   = 4096
+  guest_id = "${data.vsphere_virtual_machine.template.guest_id}"
```
```suggestion
  guest_id = data.vsphere_virtual_machine.template.guest_id
```

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
+  wait_for_guest_net_routable = false
+
+  network_interface {
+    network_id = "${data.vsphere_network.network1.id}"
```
```suggestion
    network_id = data.vsphere_network.network1.id
```

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
+  }
+
+  network_interface {
+    network_id = "${data.vsphere_network.network1.id}"
```
```suggestion
    network_id = data.vsphere_network.network1.id
```

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
+
+  disk {
+    label            = "disk0"
+    size             = "${data.vsphere_virtual_machine.template.disks.0.size}"
```
```suggestion
    size             = data.vsphere_virtual_machine.template.disks.0.size
```

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
+  disk {
+    label            = "disk0"
+    size             = "${data.vsphere_virtual_machine.template.disks.0.size}"
+    eagerly_scrub    = "${data.vsphere_virtual_machine.template.disks.0.eagerly_scrub}"
```
```suggestion
    eagerly_scrub    = data.vsphere_virtual_machine.template.disks.0.eagerly_scrub
```

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
+  }
+
+  clone {
+    template_uuid = "${data.vsphere_virtual_machine.template.id}"
```
```suggestion
    template_uuid = data.vsphere_virtual_machine.template.id
```

tenthirtyam commented on vsphere/resource_vsphere_virtual_machine_test.go:
```diff
@@ -1643,6 +1643,31 @@ func TestAccResourceVSphereVirtualMachine_cloneModifyDiskAndSCSITypeAtSameTime(t
 }
 
+func TestAccResourceVSphereVirtualMachine_cloneMultiNICWithSameGateway(t *testing.T) {
+	// This test is intentionally disabled
+	// To run it you need to configure a Windows VM to use as a template and then
+	// run Sysprep on it.
```
```suggestion
	// This test is intentionally disabled by default.
	// Requires a Windows virtual machine template to run this test.
```

spacegospod commented:
> @spacegospod - I added some minor nits to remove the legacy interpolation, but other wise this looks good and the loginc makes sense for NICs on the same network.

no problem, the rest of the VM test file is full of interpolation strings but I'm not brave enough to touch all of the tests in this change. I just fixed my own

tenthirtyam reviewed (approved):
lgtm!
